Pod/Native/NativeRunner.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  NativeRunner.swift
+//  Pods
+//
+//  Created by jacdevos on 2016/11/15.
+//
+//
+
+import Foundation
+import XCTest
+
+//Gives us the ability to run features or scenarios directly by specifying file and name
+open class NativeRunner {
+    
+    public class func runScenario(featureFile: String, scenario: String?, testCase: XCTestCase) {
+        testCase.state.loadAllStepsIfNeeded()
+
+        guard let path = (Bundle(for: type(of: testCase)).resourceURL?.appendingPathComponent(featureFile)) else {
+            XCTFail("Path could not be built for feature file: \(featureFile)")
+            return
+        }
+        
+        let features = loadFeatures(path: path)
+        
+        for feature in features {
+            let scenarios = feature.scenarios.filter({
+                scenario == nil || $0.scenarioDescription.hasPrefix(scenario!)
+            })
+            
+            if scenarios.count < 1 {
+                XCTFail("No scenario found with name: \(scenario)")
+            }
+            
+            for scenario in scenarios {
```

Comment:
This for loop is duplicate code copied from the NativeTestCase method featureScenarioTest(). I extracted the code from featureScenarioTest() into this auxiliary method in NativeTestCase so we avoid having the same code in two places:

      class func perform(scenario: NativeScenario, from feature: NativeFeature, in testCase: XCTestCase) {
        let allScenarioStepsDefined = scenario.stepDescriptions.map(testCase.state.matchingGherkinStepExpressionFound).reduce(true) { $0 && $1 }
        var allFeatureBackgroundStepsDefined = true
        
        if let defined = feature.background?.stepDescriptions.map(testCase.state.matchingGherkinStepExpressionFound).reduce(true, { $0 && $1 }) {
            allFeatureBackgroundStepsDefined = defined
        }
        
        guard allScenarioStepsDefined && allFeatureBackgroundStepsDefined else {
            XCTFail("Some step definitions not found for the scenario: \(scenario.scenarioDescription)")
            return
        }
        
        if let background = feature.background {
            background.stepDescriptions.forEach(testCase.performStep)
        }
        scenario.stepDescriptions.forEach(testCase.performStep)
    }
    
You can now replace the body of this for loop with the line:

        NativeRunner.runScenario(featureFile: featureFile, scenario: nil, testCase: testCase)

In NativeTestCase you can call this method like:

        NativeTestCase.perform(scenario: scenario, from: feature, in: self)


If you like the suggestion, please use it 🙂